frontend/front/src/pages/Public/Components/LangPrefDropdown.js

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from "react";
+import { useLocation } from "react-router-dom";
+import { Box, Menu, MenuItem, Fade, Button, Typography } from "@mui/material";
+import LanguageIcon from "@mui/icons-material/Language";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import { useTranslation } from "react-i18next";
+
+const LangPrefDropdown = () => {
+  const {
+    t,
+    i18n: { changeLanguage, language },
+  } = useTranslation();
+
+  const [anchorMore, setAnchorMore] = useState(null);
+  const [langDisplay, setLangDisplay] = useState("English");
+  const [currentLanguage, setCurrentLanguage] = useState(language);
```

Comment:
Wouldn't it be better to just use the `language` variable from line 11 rather than having this separate state variable?

Reply:
You're right; having a separate currentLanguage state might be redundant if you already have language from the useTranslation hook. Using language directly could simplify your code and ensure that your component is always in sync with the selected language.

Reply:
I fixed it!

Reply:
Maybe you didn't push the commit yet - still see it here!